### PR TITLE
Clean up import docs

### PR DESCRIPTION
**Pull request by seancolsen**
I noticed this new docs page and though it could use some cleanup. Here's what I did:

- Fix list that was not rendering due to missing empty line preceding.
- Add syntax highlighting to example JSON.
- Fix syntax errors in example JSON (trailing commas).
- Format the example JSON more idiomatically.
- Fix grammatical errors.
- Clean up language.
- Reorder JSON examples, placing "array of objects" first since it's likely to be more common.
- Give bold headings to each JSON example to make it easier for readers to skim.
- Remove `*SV` occurrences because that nomenclature may be confusing to some readers.
- Fix headings to follow proper structure.
- Write all CSV delimiters out instead of referencing python code. Some readers won't know what a "pipe" is, and will be intimidated to be dropped in a python file.
- Set custom heading fragments (e.g. `{:#csv}`).
- Simplify page title in nav menu.
- Re-order page in nav, placing it before other pages because import is commonly done early when using Mathesar.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: docs/docs/user-guide/importing-data.md [hunk 1]
```diff
@@ -1,39 +1,62 @@
...
-        "gender":"Male",
+### Header rows
+
+By default, Mathesar will use the first row of CSV data to name the columns.
+
+If you un-check **"Use first row as header"**, then Mathesar will generate default names for the columns which you can edit later.
+
+## Importing JSON data {:#json}
+
+The JSON data must be structured in of the following ways:
```
Comment (dmos62) [inline docs/docs/user-guide/importing-data.md:27]:
```suggestion
The JSON data must be structured in one of the following ways:
```

File: docs/docs/user-guide/importing-data.md [hunk 2]
```diff
@@ -1,39 +1,62 @@
...
-    "age": 21,
-    "friends": ["Bob", "Mary"]
-}
-```
+Fields in the CSV data may be delimited by any of the following characters:
 
-2. List of objects. The will create a table with columns names the same as the keys of the JSON objects. The example below creates a table with column names **first_name**, **last_name**, **gender**, **friends**, and **email** and adds two rows of data corresponding to the number of JSON objects. Missing keys in each object get __*null*__ for corresponding columns.
+| Name | Character | Notes |
+| -- | -- | -- |
+| Comma | `,` | This yields a traditional CSV file, aka "**C**omma **S**eparated **V**alue" |
```
Comment (dmos62) [inline docs/docs/user-guide/importing-data.md:13]:
```suggestion
| Comma | `,` | A traditional CSV file (a "**C**omma **S**eparated **V**alue" document) |
```

File: docs/docs/user-guide/importing-data.md [hunk 3]
```diff
@@ -1,39 +1,62 @@
...
+| Tab | _(not printable)_ | This is sometimes referred to as a _TSV_ file, for "**T**ab **S**eparated **V**alue"  |
```
Comment (dmos62) [inline docs/docs/user-guide/importing-data.md:14]:
```suggestion
| Tab | _(not printable)_ | This is sometimes referred to as a _TSV_ file (a "**T**ab **S**eparated **V**alue" document)  |
```

File: mathesar/imports/base.py [hunk 4]
```diff
@@ -5,7 +5,6 @@
 from db.tables.operations.select import get_oid_from_table
 from mathesar.errors import InvalidTableError
 
-ALLOWED_DELIMITERS = ",\t:|;"
```
Comment (dmos62) [inline mathesar/imports/base.py:8]: :+1:

File: mathesar/imports/csv.py [hunk 5]
```diff
@@ -14,6 +14,8 @@
 
 from mathesar.state import reset_reflection
 
+# The user-facing documentation replicates these delimiter characters. If you
+# change this variable, please update the documentation as well.
```
Comment (dmos62) [inline mathesar/imports/csv.py:18]: :+1:

**Review by dmos62 — COMMENTED** [review]
Thanks for the cleanup. I had some suggestions. I approve the PR. I'm not approving via the UI, because you might not notice the suggestions. Feel free to merge.

I had some reservations about putting all "separated value" formats under the CSV umbrella (feels like calling the French language a kind of English), but it is probably a useful shortcut.

File: docs/docs/user-guide/importing-data.md (hunk 1 above)
Comment (seancolsen) [reply docs/docs/user-guide/importing-data.md]: Fixed in fe76dc242661982312dacff35850f032d5d9a020

File: docs/docs/user-guide/importing-data.md (hunk 2 above)
Comment (seancolsen) [reply docs/docs/user-guide/importing-data.md]: Fixed in 8e06dbf56e3aec351505c1a8f7192f3b6f73d686

File: docs/docs/user-guide/importing-data.md (hunk 3 above)
Comment (seancolsen) [reply docs/docs/user-guide/importing-data.md]: Fixed in 8e06dbf56e3aec351505c1a8f7192f3b6f73d686

**Pull request merged**